### PR TITLE
[browser] make cwraps more conditional

### PR DESCRIPTION
**Pull request by pavelsavara**
To address feedback at https://github.com/dotnet/runtime/pull/85919#issuecomment-1540856976

### CODE REVIEW

File: src/mono/wasm/build/WasmApp.Native.targets [hunk 1]
```diff
@@ -267,6 +267,10 @@
       <EmscriptenEnvVars Include="EM_FROZEN_CACHE=True" />
       <EmscriptenEnvVars Include="DISABLE_LEGACY_JS_INTEROP=1" Condition="'$(WasmEnableLegacyJsInterop)' == 'false'" />
       <EmscriptenEnvVars Include="DISABLE_LEGACY_JS_INTEROP=0" Condition="'$(WasmEnableLegacyJsInterop)' != 'false'" />
+      <EmscriptenEnvVars Include="ENABLE_AOT_PROFILER=1" Condition="$(WasmProfilers.Contains('aot'))" />
```
Comment (kg) [inline src/mono/wasm/build/WasmApp.Native.targets:270]: I think the 1/0 could be computed expressions, but I'm fine with not doing that yet.

File: src/mono/wasm/runtime/es6/dotnet.es6.lib.js [hunk 2]
```diff
@@ -7,8 +7,10 @@
 
 // because we can't pass custom define symbols to acorn optimizer, we use environment variables to pass other build options
 const DISABLE_LEGACY_JS_INTEROP = process.env.DISABLE_LEGACY_JS_INTEROP === "1";
+const ENABLE_BROWSER_PROFILER = process.env.ENABLE_BROWSER_PROFILER === "1";
+const ENABLE_AOT_PROFILER = process.env.ENABLE_AOT_PROFILER === "1";
 
-function setup(disableLegacyJsInterop) {
+function setup(linkerDisableLegacyJsInterop, linkerEnableAotProfiler, linkerEnableBrowserProfiler) {
```
Comment (kg) [inline src/mono/wasm/runtime/es6/dotnet.es6.lib.js:13]: is there a way to make these not positional arguments, and a dict with named members instead?

File: src/mono/wasm/build/WasmApp.Native.targets (hunk 1 above)
Comment (pavelsavara) [reply src/mono/wasm/build/WasmApp.Native.targets:270]: it's a bool, and I think that MSBuild doesn't have ternary operator. Any specific hint ?

File: src/mono/wasm/wasm.proj [hunk 3]
```diff
@@ -407,8 +407,8 @@
       <CMakeBuildRuntimeCmd>cmake --build . --config $(Configuration) $(CmakeOptions)</CMakeBuildRuntimeCmd>
       <CMakeBuildRuntimeCmd Condition="'$(OS)' == 'Windows_NT'">call &quot;$(RepositoryEngineeringDir)native\init-vs-env.cmd&quot; &amp;&amp; call &quot;$([MSBuild]::NormalizePath('$(EMSDK_PATH)', 'emsdk_env.bat'))&quot; &amp;&amp; $(CMakeBuildRuntimeCmd)</CMakeBuildRuntimeCmd>
       <CMakeBuildRuntimeCmd Condition="'$(OS)' != 'Windows_NT'">bash -c 'source $(EMSDK_PATH)/emsdk_env.sh 2>&amp;1 &amp;&amp;  $(CMakeBuildRuntimeCmd)'</CMakeBuildRuntimeCmd>
-      <_DisableLegacyJsInterop Condition="'$(WasmEnableLegacyJsInterop)' == 'false'">1</_DisableLegacyJsInterop>
-      <_DisableLegacyJsInterop Condition="'$(WasmEnableLegacyJsInterop)' != 'false'">0</_DisableLegacyJsInterop>
+      <_DISABLE_LEGACY_JS_INTEROP Condition="'$(WasmEnableLegacyJsInterop)' == 'false'">1</_DISABLE_LEGACY_JS_INTEROP>
```
Comment (radical) [inline src/mono/wasm/wasm.proj:410]: Why the name change? The new one doesn't follow the general msbuild style.

File: src/mono/wasm/runtime/profiler.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ import { utf8ToString } from "./strings";
 // DumpAotProfileData stores the data into INTERNAL.aotProfileData.
 //
 export function mono_wasm_init_aot_profiler(options: AOTProfilerOptions): void {
+    mono_assert(linkerEnableAotProfiler, "AOT profiler is not enabled, please use <WasmProfilers>aot;</WasmProfilers> in your project file.");
```
Comment (radical) [inline src/mono/wasm/runtime/profiler.ts:18]: When can this happen? Can it be caught at build time too?

**Comment by pavelsavara** [comment]
CI failures are https://github.com/dotnet/runtime/issues/87414

**Review by kg — APPROVED** [review]

**Pull request merged**